cli/popper/gha.py

```diff
@@ -184,17 +185,21 @@ def run(self, action_name=None, reuse=False, parallel=False):
         self.download_actions()
         self.instantiate_runners()
 
-        if action_name:
-            self.wf.get_runner(action_name).run(reuse)
+        if self.action_name:
+            if self.with_dependencies:
```

Comment:
looks like this condition is doing the same as the outer `else` below (lines 195-196), right? Can we instead remove it?